src/app/vertextool/qgsvertexeditor.cpp

```diff
@@ -301,39 +302,68 @@ QgsVertexEditor::QgsVertexEditor(
 {
   setWindowTitle( tr( "Vertex Editor" ) );
   setObjectName( QStringLiteral( "VertexEditor" ) );
-  mTableView = new QTableView( this );
 
+  QWidget *content = new QWidget( this );
+  content->setMinimumHeight( 160 );
+  QVBoxLayout *layout = new QVBoxLayout( content );
+  layout->setContentsMargins( 0, 0, 0, 0 );
+
+  mHintLabel = new QLabel( this );
+  mHintLabel->setText( QStringLiteral( "%1\n\n%2" ).arg( tr( "Right click on the edge of an editable feature to show its table of vertices." ),
+                       tr( "When a feature is bound to this panel, dragging a rectangle to select vertices on the canvas will only select those of the bound feature." ) ) );
+  mHintLabel->setWordWrap( true );
+  mHintLabel->setAlignment( Qt::AlignHCenter | Qt::AlignVCenter );
+
+  mTableView = new QTableView( this );
   mTableView->setSelectionMode( QTableWidget::ExtendedSelection );
   mTableView->setSelectionBehavior( QTableWidget::SelectRows );
   mTableView->setItemDelegateForColumn( 0, new CoordinateItemDelegate( this ) );
   mTableView->setItemDelegateForColumn( 1, new CoordinateItemDelegate( this ) );
   mTableView->setItemDelegateForColumn( 2, new CoordinateItemDelegate( this ) );
   mTableView->setItemDelegateForColumn( 3, new CoordinateItemDelegate( this ) );
   mTableView->setItemDelegateForColumn( 4, new CoordinateItemDelegate( this ) );
+  mTableView->setVisible( false );
+
+  layout->addWidget( mTableView );
+  layout->addWidget( mHintLabel );
 
-  setWidget( mTableView );
+  setWidget( content );
 
   updateEditor( layer, selectedFeature );
 }
 
 void QgsVertexEditor::updateEditor( QgsVectorLayer *layer, QgsSelectedFeature *selectedFeature )
 {
-  delete mVertexModel;
+  if ( mSelectedFeature )
+  {
+    delete mVertexModel;
```

Comment:
I think we should also set mVertexModel to nullptr here to make sure we do not keep a dangling pointer in some case